Updates tipsy frontend for yt.4, li_ml dust grid option

anchwr opened the pull request:
Changes to tipsy2pd.py allow tipsy output files to be used with yt.4. I've also changed _gassfr to calculate the instantaneous gas SFR for tipsy outputs, since Gasoline doesn't usually calculate this in-situ, and adapted the various gadget2pd li_ml functions to work with tipsy fields. There's also a small re-ordering of pd_front_end.py so that the 'solar' variable is defined earlier, since this is called by the frontends when users select the 'rr' dust grid type.

dnarayanan commented:
these are great!  I just did a test on one of Tremmel's pioneer galaxies, and had the following notes that I was wondering if you could please update:

1. I had to update lines 138-139 in tipsy2pd to:

```        
try: cstar = cstarfac*data[("Gas","H2")][SFposs] # SF efficiency with molecular hydrogen
	except: cstar = cstarfac*data[("Gas","Mass")][SFposs]/data[("Gas","Mass")][SFposs]
```

2. quick question - was there a reason for the pd_front_end updates?

anchwr commented:
1 - Thanks! I forgot about the pioneer galaxies. I should download one so I can run tests with them as well as the Gasoline g14 galaxies. I've updated the line in question.
2 - Yes. I noticed when I ran a test with the 'rr' dust grid type option that the solar variable was being called before it was assigned, so I moved the definition up so that it would occur before the call to m_gen.

dnarayanan commented:
okay awesome thanks!  I'm in the middle of some updates to main so I might not get to merging this in until tomorrow once I sort out the conflicts on my end.
